packages/flutter/lib/src/cupertino/switch.dart

```diff
@@ -390,8 +384,10 @@ class _RenderCupertinoSwitch extends RenderConstrainedBox {
   }
 
   void _handleDragStart(DragStartDetails details) {
-    if (isInteractive)
+    if (isInteractive) {
       _reactionController.forward();
+      _emitVibration();
```

Comment:
Sorry this is becoming more complicated than initially thought. There's a bunch of nuances here it seems. On iOS, as far as I can tell, it's not possible to move enough to break the drag slop, but not enough to put the knob onto the other side and still get a knock. It may be possible here. Worth testing.

Reply:
I mean, the current implementation of the switch is wrong in respect to iOS 12. https://github.com/flutter/flutter/issues/27962 . Given that, I don't think there's really a completely 'correct' way to implement the vibration for the version of the switch that exists in flutter. No worries about the complication!